tests/unit/replicator_mock_tests.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+# Copyright (C) 2018 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+_replicator_mock_tests_
+
+replicator module - Mock unit tests for the Replicator class
+"""
+
+import mock
+import unittest
+
+from cloudant.database import CouchDatabase
+from cloudant.replicator import Replicator
+
+from tests.unit.iam_auth_tests import MOCK_API_KEY
+
+
+class ReplicatorDocumentValidationMockTests(unittest.TestCase):
+    """
+    Replicator document validation tests
+    """
+
+    def setUp(self):
+        self.repl_id = 'rep_test'
+
+        self.server_url = 'http://localhost:5984'
+        self.user_ctx = {
+            'name': 'foo',
+            'roles': ['erlanger', 'researcher']
+        }
+
+        self.source_db = 'source_db'
+        self.target_db = 'target_db'
+
+    def test_using_admin_party_source_and_target(self):
+        # setup mocks
+        m_admin_party_client = mock.MagicMock()
+        type(m_admin_party_client).admin_party = mock.PropertyMock(
+            return_value=True)
+        type(m_admin_party_client).server_url = mock.PropertyMock(
+            return_value=self.server_url)
```

Comment:
I'm wondering if we should put this code into it's own method so that it can be called by all tests in this class.  It would require 4 if blocks for options `admin_party`, `server_url`, `r_session`, and `is_iam_authenticated`.